From 24954dcfd60261600c04ffdfc9cd0b1980a4b7d3 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 3 Jan 2012 20:34:09 +0100
Subject: [PATCH] readahead: bring export definition of sd-readahead in line
 with sd-daemon (cherry picked from commit
 4f3656e1cec7fe3d7d3537e23a406cb88d734502)

---
 src/sd-readahead.c |   14 +++++++++++++-
 src/sd-readahead.h |   10 +---------
 2 files changed, 14 insertions(+), 10 deletions(-)

diff --git a/src/sd-readahead.c b/src/sd-readahead.c
index c5cfe67..a334066 100644
--- a/src/sd-readahead.c
+++ b/src/sd-readahead.c
@@ -37,6 +37,18 @@
 
 #include "sd-readahead.h"
 
+#if (__GNUC__ >= 4)
+#ifdef SD_EXPORT_SYMBOLS
+/* Export symbols */
+#define _sd_export_ __attribute__ ((visibility("default")))
+#else
+/* Don't export the symbols */
+#define _sd_export_ __attribute__ ((visibility("hidden")))
+#endif
+#else
+#define _sd_export_
+#endif
+
 static int touch(const char *path) {
 
 #if !defined(DISABLE_SYSTEMD) && defined(__linux__)
@@ -60,7 +72,7 @@ static int touch(const char *path) {
         return 0;
 }
 
-int sd_readahead(const char *action) {
+_sd_export_ int sd_readahead(const char *action) {
 
         if (!action)
                 return -EINVAL;
diff --git a/src/sd-readahead.h b/src/sd-readahead.h
index 5bf975a..ee7e306 100644
--- a/src/sd-readahead.h
+++ b/src/sd-readahead.h
@@ -56,14 +56,6 @@ extern "C" {
   See sd-readahead(7) for more information.
 */
 
-#ifndef _sd_hidden_
-#if (__GNUC__ >= 4) && !defined(SD_EXPORT_SYMBOLS)
-#define _sd_hidden_ __attribute__ ((visibility("hidden")))
-#else
-#define _sd_hidden_
-#endif
-#endif
-
 /*
   Controls ongoing disk read-ahead operations during boot-up. The argument
   must be a string, and either "cancel", "done" or "noreplay".
@@ -72,7 +64,7 @@ extern "C" {
   done = terminate read-ahead data collection, keep collected information
   noreplay = terminate read-ahead replay
 */
-int sd_readahead(const char *action) _sd_hidden_;
+int sd_readahead(const char *action);
 
 #ifdef __cplusplus
 }
